build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.nexus-publish.gradle.kts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("com.hedera.hashgraph.lifecycle")
+    id("io.github.gradle-nexus.publish-plugin")
+}
+
+nexusPublishing {
+    repositories {
+        sonatype {
+            username = System.getenv("OSSRH_USERNAME")
+            password = System.getenv("OSSRH_PASSWORD")
+        }
+    }
+}
+
+tasks.named("closeSonatypeStagingRepository") {
+    // The publishing of all components to Maven Central is automatically done before close (which
+    // is done before release).
+    dependsOn(subprojects.map { ":${it.name}:releaseMavenCentral" })
```

Comment:
This is the important "tweak" - all projects are published before releasing

Following what we did and tested here: https://github.com/hashgraph/pbj/blob/main/pbj-core/buildSrc/src/main/kotlin/com.hedera.pbj.root.gradle.kts#L41-L45